furniture_bench/envs/furniture_sim_env.py

```diff
@@ -1088,15 +1125,41 @@ def _reset_franka(self, env_idx, dof_pos=None):
         self.dof_vel[:, 0 : self.franka_num_dofs] = torch.tensor(
             [0] * len(self.default_dof_pos), device=self.device, dtype=torch.float32
         )
-        idxs = torch.tensor(self.franka_handles, device=self.device, dtype=torch.int32)[
-            env_idx : env_idx + 1
-        ]
 
+        actor_idx = self.franka_actor_idxs_all_t[env_idx].reshape(1, 1)
         self.isaac_gym.set_dof_state_tensor_indexed(
             self.sim,
             gymtorch.unwrap_tensor(self.dof_states),
-            gymtorch.unwrap_tensor(idxs),
-            len(idxs),
+            gymtorch.unwrap_tensor(actor_idx),
+            len(actor_idx),
+        )
+
+    def _reset_franka_all(self, dof_pos=None):
+        # Low randomness only.
+        if self.from_skill >= 1:
+            dof_pos = torch.from_numpy(self.default_dof_pos)
+            ee_pos = torch.from_numpy(
+                self.furniture.furniture_conf["ee_pos"][self.from_skill]
+            )
+            ee_quat = torch.from_numpy(
+                self.furniture.furniture_conf["ee_quat"][self.from_skill]
+            )
+            dof_pos = self.robot_model.inverse_kinematics(ee_pos, ee_quat)
+        else:
+            dof_pos = self.default_dof_pos if dof_pos is None else dof_pos
+
+        self.dof_pos[:, 0 : self.franka_num_dofs] = torch.tensor(
+            dof_pos, device=self.device, dtype=torch.float32
+        )
+        self.dof_vel[:, 0 : self.franka_num_dofs] = torch.tensor(
+            [0] * len(self.default_dof_pos), device=self.device, dtype=torch.float32
+        )
```

Comment:
Could you refactor this block into a method? That way, both `_reset_franka_all` and `_reset_franka` can utilize it.